core/src/impl/Kokkos_Core.cpp

```diff
@@ -373,6 +372,21 @@ void initialize_internal(const InitArguments& args) {
               << std::endl;
   }
 #endif
+}
+
+void pre_initialize(const InitArguments& args) {
+  if (args.disable_warnings) g_show_warnings = false;
+}
+
+void core_initialize(const InitArguments& args) { initialize_backends(args); }
```

Comment:
I don't see the point of this function.  Do we anticipate ever having anything else than initializing all the backends inside it?  Also I would only expose `pre` and `post`.